virtualizarr/tests/test_readers/test_hdf_integration.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+import xarray as xr
+import xarray.testing as xrt
+
+import virtualizarr
+from virtualizarr.backend import FileType
+from virtualizarr.readers.hdf import HDFVirtualBackend
+from virtualizarr.tests import requires_kerchunk
+
+
+@requires_kerchunk
+@pytest.mark.parametrize("hdf_backend", [None, HDFVirtualBackend])
+class TestIntegration:
+    @pytest.mark.xfail(reason="0 time start is being interpreted as fillvalue")
+    def test_filters_h5netcdf_roundtrip(
+        self, tmpdir, filter_encoded_roundtrip_hdf5_file, hdf_backend
+    ):
+        ds = xr.open_dataset(filter_encoded_roundtrip_hdf5_file, decode_times=True)
+        vds = virtualizarr.open_virtual_dataset(
+            filter_encoded_roundtrip_hdf5_file,
+            loadable_variables=["time"],
+            cftime_variables=["time"],
+            backend=hdf_backend,
+        )
+        kerchunk_file = f"{tmpdir}/kerchunk.json"
+        vds.virtualize.to_kerchunk(kerchunk_file, format="json")
+        roundtrip = xr.open_dataset(kerchunk_file, engine="kerchunk", decode_times=True)
+        xrt.assert_allclose(ds, roundtrip)
+
+    @pytest.mark.xfail(reason="Coordinate issue affecting kerchunk and HDF reader.")
```

Comment:
I've included the issue for the xfailed test for CF time.  I haven't created an issue for the coordinate dim xfail yet since this is actually specific to this branch.  This is related to https://github.com/zarr-developers/VirtualiZarr/pull/260 and the way kerchunk serializes 0d hdf datasets though I don't fully understand the mechanics.  Once this branch is merged I can investigate a bit more and file a complete issue.